File: src/main/java/net/reflact/client/hud/ReflactHud.java
package net.reflact.client.hud;

import net.fabricmc.fabric.api.client.rendering.v1.HudRenderCallback;
import net.reflact.client.hud.overlay.*;

public class ReflactHud {

    public static void initialize() {
        OverlayManager.INSTANCE.register(new HealthBarOverlay());
        OverlayManager.INSTANCE.register(new ManaBarOverlay());
        OverlayManager.INSTANCE.register(new HotbarOverlay());
        OverlayManager.INSTANCE.register(new InfoHudOverlay());
        OverlayManager.INSTANCE.register(new ArmorHudOverlay());
        OverlayManager.INSTANCE.register(new StatusEffectOverlay());
        OverlayManager.INSTANCE.register(new TargetHudOverlay());
        OverlayManager.INSTANCE.register(new SpellHudOverlay());
        
        // Notifications
        OverlayManager.INSTANCE.register(new net.reflact.client.hud.overlay.ReflactOverlay() {
            @Override public void render(net.minecraft.client.gui.DrawContext context, float tickDelta) {
                net.minecraft.client.MinecraftClient client = net.minecraft.client.MinecraftClient.getInstance();
                int width = client.getWindow().getScaledWidth();
                int height = client.getWindow().getScaledHeight();
                net.reflact.client.ui.NotificationManager.render(context, client.textRenderer, width, height);
            }
            @Override public int getX() { return 0; }
            @Override public int getY() { return 0; }
            @Override public void setX(int x) {}
            @Override public void setY(int y) {}
            @Override public int getWidth() { return 0; }
            @Override public int getHeight() { return 0; }
            @Override public boolean isEnabled() { return true; }
            @Override public String getName() { return "Notifications"; }
        });

        HudRenderCallback.EVENT.register(OverlayManager.INSTANCE::onHudRender);
    }
}

File: src/main/kotlin/net/reflact/client/hud/ReflactHud.kt
package net.reflact.client.hud

import net.fabricmc.fabric.api.client.rendering.v1.HudRenderCallback
import net.minecraft.client.MinecraftClient
import net.minecraft.client.gui.DrawContext
import net.reflact.client.hud.overlay.*
import net.reflact.client.ui.NotificationManager

object ReflactHud {

    fun initialize() {
        OverlayManager.INSTANCE.register(HealthBarOverlay())
        OverlayManager.INSTANCE.register(ManaBarOverlay())
        OverlayManager.INSTANCE.register(HotbarOverlay())
        OverlayManager.INSTANCE.register(InfoHudOverlay())
        OverlayManager.INSTANCE.register(ArmorHudOverlay())
        OverlayManager.INSTANCE.register(StatusEffectOverlay())
        OverlayManager.INSTANCE.register(TargetHudOverlay())
        OverlayManager.INSTANCE.register(SpellHudOverlay())
        OverlayManager.INSTANCE.register(MinimapOverlay())

        // Notifications
        OverlayManager.INSTANCE.register(object : ReflactOverlay {
            override fun render(context: DrawContext, tickDelta: Float) {
                val client = MinecraftClient.getInstance()
                val width = client.window.scaledWidth
                val height = client.window.scaledHeight
                NotificationManager.render(context, client.textRenderer, width, height)
            }

            override var x: Int
                get() = 0
                set(value) {}
            override var y: Int
                get() = 0
                set(value) {}
            override var width: Int get() = 0; set(value) {}
            override var height: Int get() = 0; set(value) {}
            override val isEnabled: Boolean get() = true
            override val name: String get() = "Notifications"
        })

        HudRenderCallback.EVENT.register(OverlayManager.INSTANCE::onHudRender)
    }
}
